tools/python-integration-tests/test.py

```diff
@@ -0,0 +1,378 @@
+              "destroy",
+              self.deployment_name,
+              "--auto-approve"
+          ]
+
+        subprocess.run(cmd, check=True, cwd=self.workspace)
+        os.remove(f"{self.deployment_name}.tgz")
+        shutil.rmtree(self.deployment_name)
+
+
+class Test:
+    def __init__(self, ssh, deployment):
+        self.ssh = ssh
+        self.deployment = deployment
+        self.job_list = {}
+
+    def get_directory_contents(self):
+        stdin, stdout, stderr = self.ssh.exec_command('ls -l')
+        return stdout.read().decode()
+
+    def get_sinfo(self):
+        stdin, stdout, stderr = self.ssh.exec_command('sinfo')
+        return stdout.read().decode()
+
+    def print_jobs(self):
+        for job in self.job_list.items():
+            print(job)
+        print()
+
+    def print_job_output(self, job_id: str):
+        stdin, stdout, stderr = self.ssh.exec_command(f"cat slurm-{job_id}.out")
+        print(f"slurm-{job_id}.out:", stdout.read().decode())
+
+    def get_slurm_topology(self):
+        stdin, stdout, stderr = self.ssh.exec_command("scontrol show topo")
+        return stdout.read().decode() 
+
+    def monitor_squeue(self):
+        # Monitors squeue and updates self.job_list until all running jobs are complete.
+        lines = []
+
+        while True:
+            stdin, stdout, stderr = self.ssh.exec_command('squeue')
+
+            lines = stdout.read().decode().splitlines()[1:] # Skip header
+
+            if not lines:
+                break
+            for line in lines:
+                parts = line.split()
+                job_id, partition, _, _, state, times, nodes, nodelist = line.split()
+
+                if job_id not in self.job_list:
+                    print(f"Job id {job_id} is not recognized.")
+                else:
+                    self.job_list[job_id].update({
+                        "partition": partition,
+                        "state": state,
+                        "time": times,
+                        "nodes": nodes,
+                        "nodelist": nodelist,
+                    })
+            time.sleep(5)
+
+    def is_job_complete(self, job_id: str):
+        # Checks if a job successfully completed.
+        stdin, stdout, stderr = self.ssh.exec_command(f'scontrol show job {job_id}')
+        pattern = r"JobState=(\w+)"
+        match = re.search(pattern, stdout.read().decode())
+        if match:
+            return match.group(1) == "COMPLETED"
+        return False
+
+    def submit_job(self, cmd: str):
+        stdin, stdout, stderr = self.ssh.exec_command(cmd)
+        jobID = stdout.read().decode().split()[-1]
+        self.job_list[jobID] = {}
+
+    def get_node_depth(self, switch_name: str):
+        return switch_name.count("_")
+
+    def get_real_rack(self, node: str):
+        result = run_command(f"gcloud compute instances describe {node} --zone={self.deployment.zone} --project={self.deployment.project_id} --flatten='resourceStatus:*'")
```

Comment:
`s/--flatten='resourceStatus:*'"/--format='value(resourceStatus.physicalHost)'` to avoid tricky parsing that would fail if new field is added to the `resourceStatus`

Reply:
Updated!